### PR TITLE
Permit configuration for noncallable methods

### PR DESCRIPTION
**Pull request by geigerj**
This change corresponds to a change in the codegen removing
"callable" versions of generated API methods. The standard
versions of generated methods may now be passed a CallOptions
object to perform the same configuration that previously had been
done through the callable methods.

- Revert renaming of ApiCallableDefault from CallOptions, which
  conflicts with the new CallOptions object
- Add CallOptions object
- Add global value to indicate that a particular setting should
  be inherited from defaults
- Add testing for CallOptions

### CODE REVIEW

File: test/test_api_callable.py [hunk 1]
```diff
@@ -115,27 +115,68 @@ def grpc_return_value(request, *dummy_args, **dummy_kwargs):
...
+
+    def test_call_options_update_none(self):
+        options = api_callable.CallOptions(timeout=23, page_streaming=False)
+        options.update(None)
+        self.assertEqual(options.timeout, 23)
+        self.assertEqual(options.is_retrying, api_callable.OPTION_INHERIT)
+        self.assertFalse(options.page_streaming)
+        self.assertEqual(options.max_attempts, api_callable.OPTION_INHERIT)
+
+    def test_call_option_normalize(self):
```
Comment (tbetbetbe) [inline test/test_api_callable.py]: minor: s/test_call_option/test_call_options/  for consistency with the other test methods

File: test/test_api_callable.py [hunk 2]
```diff
@@ -115,27 +115,68 @@ def grpc_return_value(request, *dummy_args, **dummy_kwargs):
...
+        options = api_callable.CallOptions(timeout=23, is_retrying=True)
+        options.normalize()
+        self.assertEqual(options.timeout, 23)
+        self.assertTrue(options.is_retrying)
+        self.assertIsNone(options.page_streaming)
+        self.assertIsNone(options.max_attempts)
+
+    def test_call_option_normalize_default(self):
```
Comment (tbetbetbe) [inline test/test_api_callable.py]: minor: s/test_call_option/test_call_options/  for consistency with the other test methods

Comment (geigerj) [inline test/test_api_callable.py]: Done.

File: test/test_api_callable.py (hunk 1 above)
Comment (geigerj) [inline test/test_api_callable.py]: Done.

**Comment by tbetbetbe** [comment]
LGTM

**Pull request merged**